shared/constants/metametrics.ts

```diff
@@ -0,0 +1,637 @@
+  KeyBatchAccountBalanceRequests = 'Key Batch account balance requests',
+  MetricsOptIn = 'Metrics Opt In',
+  MetricsOptOut = 'Metrics Opt Out',
+  NavAccountMenuOpened = 'Account Menu Opened',
+  NavAccountDetailsOpened = 'Account Details Opened',
+  NavConnectedSitesOpened = 'Connected Sites Opened',
+  NavMainMenuOpened = 'Main Menu Opened',
+  NavNetworkMenuOpened = 'Network Menu Opened',
+  NavSettingsOpened = 'Settings Opened',
+  NavAccountSwitched = 'Account Switched',
+  NavNetworkSwitched = 'Network Switched',
+  NavBuyButtonClicked = 'Buy Button Clicked',
+  NavSendButtonClicked = 'Send Button Clicked',
+  NavSwapButtonClicked = 'Swap Button Clicked',
+  SrpToConfirmBackup = 'SRP Backup Confirm Displayed',
+  WalletSetupStarted = 'Wallet Setup Selected',
+  WalletSetupCanceled = 'Wallet Setup Canceled',
+  WalletSetupFailed = 'Wallet Setup Failed',
+  WalletCreated = 'Wallet Created',
+  NftAdded = 'NFT Added',
+  OnrampProviderSelected = 'On-ramp Provider Selected',
+  PermissionsApproved = 'Permissions Approved',
+  PermissionsRejected = 'Permissions Rejected',
+  PermissionsRequested = 'Permissions Requested',
+  PhishingPageDisplayed = 'Phishing Page Displayed',
+  PortfolioLinkClicked = 'Portfolio Link Clicked',
+  PublicAddressCopied = 'Public Address Copied',
+  ProviderMethodCalled = 'Provider Method Called',
+  SignatureApproved = 'Signature Approved',
+  SignatureFailed = 'Signature Failed',
+  SignatureRejected = 'Signature Rejected',
+  SignatureRequested = 'Signature Requested',
+  TokenImportButtonClicked = 'Import Token Button Clicked',
+  TokenScreenOpened = 'Token Screen Opened',
+  SupportLinkClicked = 'Support Link Clicked',
+  TokenAdded = 'Token Added',
+  TokenDetected = 'Token Detected',
+  TokenHidden = 'Token Hidden',
+  TokenImportCanceled = 'Token Import Canceled',
+  TokenImportClicked = 'Token Import Clicked',
+  OnboardingWelcome = 'App Installed',
+  OnboardingWalletCreationStarted = 'Wallet Setup Selected',
+  OnboardingWalletImportStarted = 'Wallet Import Started',
+  OnboardingWalletCreationAttempted = 'Wallet Password Created',
+  OnboardingWalletSecurityStarted = 'SRP Backup Selected',
+  OnboardingWalletSecuritySkipInitiated = 'SRP Skip Backup Selected',
+  OnboardingWalletSecuritySkipConfirmed = 'SRP Backup Skipped',
+  OnboardingWalletSecuritySkipCanceled = 'SRP Skip Backup Canceled',
+  OnboardingWalletSecurityPhraseRevealed = 'SRP Revealed',
+  OnboardingWalletSecurityPhraseWrittenDown = 'SRP Backup Confirm Display',
+  OnboardingWalletSecurityPhraseConfirmed = 'SRP Backup Confirmed',
+  OnboardingWalletCreationComplete = 'Wallet Created',
+  OnboardingWalletSetupComplete = 'Application Opened',
+  OnboardingWalletAdvancedSettings = 'Settings Updated',
+  OnboardingWalletImportAttempted = 'Wallet Import Attempted',
+  OnboardingWalletVideoPlay = 'SRP Intro Video Played',
+  OnboardingTwitterClick = 'External Link Clicked',
+}
+
+export enum MetaMetricsEventAccountType {
```

Comment:
Note that previously all of these event-related enums were packaged into one object. Enum values must be simple, so we can't nest these into a larger enum.